### ISSUE TITLE
Plugin sometimes ingests dozens of duplicates for kube-inventory data

### ISSUE THREAD

**Issue by v1r7u** [splunk/fluent-plugin-kubernetes-objects#39]
**What happened**: `kubernetes_objects` plugin sometimes ingests into Elasticsearch a few times more data. For example, instead of reporting the state of 20 Kubernetes nodes, it sends 80.
For example, at the screenshot is amount of kube-node objects ingested by the plugin. We didn't do any cluster scaling or maintenance over this period: all these time we had exactly the same amount of nodes:
![image](https://user-images.githubusercontent.com/1583399/65599569-7444bf00-df9e-11e9-8c39-50fa1763c4ba.png)

The same issue happens with pod and service objects.

**What you expected to happen**: the plugin sends only unique data withing specified in configuration period

**How to reproduce it (as minimally and precisely as possible)**:
A part of fluentd config, dedicated to node objects:
```
    <source>
      @id kube_nodes
      @type kubernetes_objects
      tag k8s.objects.nodes

      <pull>
        resource_name nodes
      </pull>
    </source>
...
    <match k8s.objects.nodes>
      @id elasticsearch.k8s.nodes
      @type elasticsearch
      @log_level info
      type_name fluentd
      include_tag_key true
      host elasticsearch-logging
      port 9200
      logstash_format true
      logstash_prefix k8s-nodes
      logstash_dateformat %Y-%m-%d
      reload_connections true
      reload_after 50
      reload_on_failure true
      reconnect_on_error true
      prefer_oj_serializer true
      <buffer>
        @type file
        path /var/log/fluentd-buffers/k8s.nodes.buffer
        chunk_limit_size 2M
        total_limit_size 20M
        overflow_action block
        flush_mode interval
        flush_thread_count 2
        flush_interval 1m
        retry_type exponential_backoff
        retry_forever
        retry_max_interval 60
      </buffer>
    </match>
```

**Anything else we need to know?**: We run the service in Kubernetes, using default service account and ingest data directly to Elasticsearch, hosted in the same namespace

**Environment**:
- Kubernetes version (use `kubectl version`): `Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}`
- Ruby version (use `ruby --version`): `ruby 2.3.3p222 (2016-11-21) [x86_64-linux-gnu]`
- `'fluentd', '<=1.6.3'`, `'fluent-plugin-kubernetes-objects', '~>1.1.2'`, and `'fluent-plugin-elasticsearch', '~>3.5.3'`
- OS (e.g: `cat /etc/os-release`): `Debian GNU/Linux 9 (stretch)` from docker image `debian:stretch-20190812-slim`
- Elasticsearch version: `6.2.4`

**Comment by matthewmodestino** [comment]
Hi v1r7u!

Thanks for checking out our plugin!

Do you have any analysis/logs that show there are duplicates passed by the input plugin to the output plugin you are using? 

Have you tried analyzing with a more verbose logging level or the fluentd stdout filter?



We do not support your chosen output plugin, but I’d recommend you check whether any sort of “acks” are used that could cause retransmissions.

You could also use the copy output plugin and send to a splunk trial u can run in k8s, see if you see the same thing :)

**Comment by v1r7u** [comment]
Excuse me for the late response.
I do not know how it works, but after creating the issue here - the problem had gone in all our clusters :see_no_evil:
The issue is created 24 days ago, last spike - ~20 days ago.
The only one thing I did for sure - switched off watching kube-events.

I had to investigate logs in more detail at the very beginning - thanks for the reminder :wink: If I see the issue again, I'll gather some logs for sure.
To be honest, I'm not eager to debug the issue further - if it works, let it be
